crates/lsp/src/lib.rs

```diff
@@ -1,11 +1,8 @@
-// TODO: Remove this
-#![allow(dead_code)]
```

Comment:
I removed this in favor of a few local `#[allow(dead_code)]`s, or just removing some dead things entirely if they looked outdated.

I'm very thankful to reenable dead code analysis, because it really helps me remember to remove things I didn't end up using